### PR TITLE
Gitlab pipelines: add a small legend in the logs to interpret "[x]"

### PR DESCRIPTION
**Pull request by alalazo**

### CODE REVIEW

**Review by zackgalbreath — DISMISSED** [review]

File: lib/spack/spack/ci.py [hunk 1]
```diff
@@ -315,18 +315,16 @@ def print_staging_summary(spec_labels, dependencies, stages):
         return
 
     tty.msg('  Staging summary:')
```
Comment (adrienbernede) [inline lib/spack/spack/ci.py]: If @scottwittenburg argues that we need to list all the jobs, including those not needing a rebuild, then the minimum we could do is to add some sort of legend here.
```
tty.msg('  Staging summary: ([x] -> needs rebuild)')
```

Note that this is no longer true with the current PR proposal, which is to only list specs that will be rebuilt.

**Review by adrienbernede — DISMISSED** [review]
If the consensus is that we only need to know the jobs that will be rebuilt, then this proposal removes any ambiguity.

**Review by scottwittenburg — CHANGES_REQUESTED** [review]
In the past it has been helpful for me to be able to see the specs/hashes that were considered, as well as which ones were actually scheduled, as provided by the existing implementation.  To remove the ambiguity, could we just print a message at the top of the staging summary saying "An `x` indicates specs that need to be rebuilt"?

**Review by scottwittenburg — APPROVED** [review]
Looks great, thank you!

**Pull request merged**